### PR TITLE
New Presence: Invidious

### LINKED ISSUE #1883: Invidious
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** Xpert#6337

**What's the name of the service?**  Invidious | All instances are at this [link](https://github.com/iv-org/invidious/wiki/Invidious-Instances)

**What should the Presence display?** It should be exactly like the Youtube presence (Title, Channel Name, Time Remaining)
**If possible, please provide a logo for the service (512x512 minimum)** 
maxresdefault

### PR DESCRIPTION
**Pull request by TheLastZombie**
Adds support for [Invidious](https://github.com/iv-org/invidious/).

Invidious can be self-hosted. There used to be an [official instance](https://invidio.us/), but it was since replaced with a [list of popular public instances](https://instances.invidio.us/). URLs were taken from said list, minus .onion and .i2p domains.

### Screenshots

image
image
image

Edit: Closes #1883, thanks @Isa27-0 for the heads-up.

### CODE REVIEW

File: websites/I/Invidious/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "invidious.site",
+    "invidious.snopyta.org",
+    "invidious.toot.koeln",
+    "invidious.tube",
+    "invidious.xyz",
+    "vid.mint.lgbt",
+    "yewtu.be"
+  ],
+  "version": "1.0.0",
+  "logo": "https://invidio.us/static/img/invidious-logo-dark.svg",
```
Comment (Bas950) [inline websites/I/Invidious/dist/metadata.json:26]: upload to imgur

**Review by Bas950 — CHANGES_REQUESTED** [review]
Add support for pages other then /watch
As it might look broken on the other pages.

**Comment by TheLastZombie** [comment]
New screenshots:

|  |  |
|-|-|
|  |  |
|  |  |
|  |  |

File: websites/I/Invidious/presence.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    case "channel":
+      presenceData.details = "Viewing channel";
+      presenceData.state = document.getElementsByClassName("channel-profile")[0].textContent.trim();
+      break;
+
+    default:
+      presence.setTrayTitle();
+      presenceData = undefined;
+      break;
```
Comment (Bas950) [inline websites/I/Invidious/presence.ts:77]: dont set it to undefined.

Comment (Bas950) [reply websites/I/Invidious/presence.ts:77]: it can actually cause issues.

**Review by Bas950 — CHANGES_REQUESTED** [review]
Add support for pages other then /watch
As it might look broken on the other pages.

**Review by Bas950 — APPROVED** [review]
Add support for pages other then /watch
As it might look broken on the other pages.

**Review by fr0r — APPROVED** [review]
test

**Review by Timeraa — APPROVED** [review]

**Pull request merged**